### PR TITLE
Clarify documentation on series endpoint

### PR DESCRIPTION
**Pull request by kubicgruenfeld**
**What this PR does**:

Clarify documentation about the `/series` endpoint and adds a note about queries without `__name__` label to ingesters.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@
     - `-experimental.tsdb.stripe-size` changed to `-experimental.blocks-storage.tsdb.stripe-size`
     - `-experimental.tsdb.wal-compression-enabled` changed to `-experimental.blocks-storage.tsdb.wal-compression-enabled`
     - `-experimental.tsdb.flush-blocks-on-shutdown` changed to `-experimental.blocks-storage.tsdb.flush-blocks-on-shutdown`
+* [CHANGE] Clarify limitations to the `/series` endpoint. #2953
```
Comment (pracucci) [inline CHANGELOG.md:43]: Any documentation update should be an `[ENHANCEMENT]`. Changes are application-level behavioural changes.

File: docs/configuration/v1-guarantees.md [hunk 2]
```diff
@@ -23,7 +23,8 @@ The Cortex maintainers commit to ensuring future version of Cortex can read data
 
 Cortex strives to be 100% API compatible with Prometheus (under `/api/prom/*`); any deviation from this is considered a bug, except:
 
-- Requiring the `__name__` label on queries.
+- Requiring the `__name__` label on queries (except for queries to Ingesters).
```
Comment (pracucci) [inline docs/configuration/v1-guarantees.md:26]:
```suggestion
- Requiring the `__name__` label on queries when querying the chunks storage (queries to ingesters or clusters running the blocks storage are not affected).
```

File: docs/configuration/v1-guarantees.md [hunk 3]
```diff
@@ -23,7 +23,8 @@ The Cortex maintainers commit to ensuring future version of Cortex can read data
...
+- For queries to the `/series` endpoint given time ranges older than the data stored in Ingesters will be ignored.
```
Comment (pracucci) [inline docs/configuration/v1-guarantees.md:27]: Few comments:
- I would name the API `/api/v1/series`
- This also affects `/api/v1/labels` and `/api/v1/label/{name}/values`
- Maybe it's more clear if we say that the API endpoints only query data from ingesters (not long-term storage)

File: docs/configuration/v1-guarantees.md (hunk 2 above)
Comment (kubicgruenfeld) [reply docs/configuration/v1-guarantees.md]: Thanks, used the suggested change.

File: CHANGELOG.md (hunk 1 above)
Comment (kubicgruenfeld) [reply CHANGELOG.md]: Moved to [ENHANCEMENT]

File: docs/configuration/v1-guarantees.md (hunk 3 above)
Comment (kubicgruenfeld) [reply docs/configuration/v1-guarantees.md]: I have clarified the API path and added the mentioned endpoints. Also added that data is only retrieved from ingesters, wdyt?

File: CHANGELOG.md [hunk 4]
```diff
@@ -87,6 +87,7 @@
 * [ENHANCEMENT] Flusher: Added `-flusher.exit-after-flush` option (defaults to true) to control whether Cortex should stop completely after Flusher has finished its work. #2877
 * [ENHANCEMENT] Added metrics `cortex_config_hash` and `cortex_runtime_config_hash` to expose hash of the currently active config file. #2874
 * [ENHANCEMENT] Logger: added JSON logging support, configured via the `-log.format=json` CLI flag or its respective YAML config option. #2386
+* [ENHANCEMENT] Clarify limitations to the `/api/v1/series` endpoint. #2953
```
Comment (pracucci) [inline CHANGELOG.md:90]:
```suggestion
* [ENHANCEMENT] Clarify limitations of the `/api/v1/series`, `/api/v1/labels` and `/api/v1/label/{name}/values` endpoints. #2953
```

File: docs/configuration/v1-guarantees.md [hunk 5]
```diff
@@ -23,7 +23,8 @@ The Cortex maintainers commit to ensuring future version of Cortex can read data
 
 Cortex strives to be 100% API compatible with Prometheus (under `/api/prom/*`); any deviation from this is considered a bug, except:
 
-- Requiring the `__name__` label on queries.
+- Requiring the `__name__` label on queries when querying the chunks storage (queries to ingesters or clusters running the blocks storage are not affected).
+- For queries to the `/api/v1/series`, `/api/v1/labels` and `/api/v1/label/{name}/values` endpoints, given time ranges older than the data stored in ingesters will be ignored and data will only retrieved from ingesters.
```
Comment (pracucci) [inline docs/configuration/v1-guarantees.md:27]:
```suggestion
- For queries to the `/api/v1/series`, `/api/v1/labels` and `/api/v1/label/{name}/values` endpoints, query's time range is ignored and the data is always fetched from ingesters.
```

File: docs/configuration/v1-guarantees.md (hunk 3 above)
Comment (pracucci) [reply docs/configuration/v1-guarantees.md]: LGTM, thanks!

**Pull request merged**